airbyte-integrations/connectors/source-gitlab/source_gitlab/auth.py

```diff
@@ -0,0 +1,36 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from typing import Any, Mapping
+
+from airbyte_cdk.sources.streams.http.auth import HttpAuthenticator
+
+
+class GitlabAuthenticator(HttpAuthenticator):
+    def __init__(self, token: str, auth_header: str = "Private-Token"):
+        self.auth_header = auth_header
+        self._token = token
+
+    def get_auth_header(self) -> Mapping[str, Any]:
+        return {self.auth_header: self._token}
```

Comment:
Looks like this. Maybe we could update `TokenAuthenticator` class to take additional parameter `delimeter` which will be ` ` (space) by default. This update will remove current `GitlabAuthenticator` class.

Reply:
@vitaliizazmic Yes, you are right, I didn’t use TokenAuthenticator because it will use additional problems in the header parameter, which will break the authorization.
@Zirochkaa, great idea, I think you can create issue about it.

Reply:
We can use strip() instead of creating new parameters in TokenAuthenticator.